apps/E2E/src/Menu/pages/MenuPageObject.ts

```diff
@@ -57,26 +77,39 @@ class MenuPageObject extends BasePage {
     await (await this.getMenuComponentSelector(menuComponentSelector)).addValue(key);
   }
 
+  async clickComponent(menuComponentSelector?: MenuComponentSelector): Promise<void> {
+    const selector = menuComponentSelector ?? MenuComponentSelector.PrimaryComponent;
+    await (await this.getMenuComponentSelector(selector)).click();
+  }
+
   /* Returns the correct WebDriverIO element from the Button Selector */
   async getMenuComponentSelector(menuComponentSelector?: MenuComponentSelector): Promise<WebdriverIO.Element> {
```

Comment:
If this only returns MenuItem's, let's change the name to `getMenuItemComponentSelector` to better represent that. Helps readability.

Reply:
`getMenuComponentSelector` also returns the `MenuTrigger` element that opens the menu. I've left the name the same right now, but should these be two methods?

Reply:
Where does it return a MenuTrigger item below? On first glance, it looks like it doesn't. However, now I see `_primaryComponent` does. Is there any particular reason it's under `default:`? 

To improve readability, let's add:
`case MenuComponentSelector.MenuTrigger: return await this._primaryComponent;`